ethereum/ethereum_connector.bal

```diff
@@ -0,0 +1,1615 @@
+// Copyright (c) 2018 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/io;
+
+function EthereumConnector::getWeb3ClientVersion() returns Result|EthereumError {
+    endpoint http:Client httpClient = self.httpClient;
+    var httpResponse = httpClient -> post("/", constructRequest(self.jsonRpcVersion, check <int>self.networkId,
+            WEB3_CLIENT_VERSION , EMPTY_PARAMS));
+    match httpResponse {
+        error err => {
+            EthereumError ethereumError = {};
```

Comment:
Change the function signature to return `error` and just return err here. Check all the places.